python/revng/daemon/graphql.py

```diff
@@ -41,10 +43,30 @@ def run_in_executor(function: Callable[P, T], *args: P.args, **kwargs: P.kwargs)
     return loop.run_in_executor(executor, partial(function, *args, **kwargs))
 
 
+@dataclass
+class Diff:
+    diff: str
+
+
+@dataclass
+class Produced:
+    result: str
```

Comment:
What's the reason for these guys to exist?

Reply:
graphql `union`s can only be done between structs and not basic types (e.g. I cannot write `AnalysisResult = string | SimpleError | DocumentError`) so those classes are needed as a dummy struct to package the string.